Source/Programs/wavm.cpp

```diff
@@ -0,0 +1,224 @@
+#include "Core/Core.h"
+#include "Core/MemoryArena.h"
+#include "Core/Platform.h"
+#include "AST/AST.h"
+#include "Runtime/Runtime.h"
+#include "Runtime/RuntimePrivate.h"
+
+#include "CLI.h"
+
+bool endsWith(const char *str, const char *suffix)
+{
+	if(!str || !suffix) { return false; }
+	size_t lenstr = strlen(str);
+	size_t lensuffix = strlen(suffix);
+	if(lenstr < lensuffix) { return false; }
+	return (strncmp(str+lenstr-lensuffix, suffix, lensuffix) == 0);
+}
+
+void showHelp()
+{
+	std::cerr << "Usage: wavm [switches] [programfile] [--] [arguments]" << std::endl;
+	std::cerr << "  --text in.wast\t\tSpecify text program file (.wast)" << std::endl;
+	std::cerr << "  --binary in.wasm in.js.mem\tSpecify binary program file (.wasm) and memory file (.js.mem)" << std::endl;
+	std::cerr << "  -f|--function name\t\tSpecify function name to run in module rather than main" << std::endl;
+	std::cerr << "  -c|--check\t\t\tCheck program syntax" << std::endl;
+	std::cerr << "  --outputAST file\t\tOutput module AST to file" << std::endl;
+	std::cerr << "  --\t\t\t\tStop parsing arguments" << std::endl;
+}
+
+int main(int argc,char** argv)
+{
+	bool onlyValidate = false;
+	const char* sourceFile = 0;
+	const char* binaryFile = 0;
+	const char* binaryMemFile = 0;
+	auto functionName = "main";
+	const char* outputASTFilename = 0;
+
+	auto args = argv;
+	while (*++args) {
```

Comment:
Sorry about that, I must say this is the first time I have ever seen this brace style and while I have been trying to follow it, habits die hard and feel free to point any out.  Any easy way to add a style cop to travis?